root/etc/systemd/system/discovery-autostart.service

```diff
@@ -1,7 +1,8 @@
 [Unit]
 Description=Start downloaded extension scripts
+Requires=nm-connected.service
```

Comment:
Requires= is critical to ensure all the discovery-….service are not started before nm-connected.service finished **successfully**. Without Requires= (only with After=) the services would start regardless of nm-online's return code.